### ISSUE TITLE
NAT Port Forward Alias not working

### ISSUE THREAD

**Issue by itheiss** [opnsense/core#2051]
Hi,

after upgrading to OPNsense 18.1.b_273-amd64 my NAT Port Forward rules stopped working. All traffic suddenly where blocked by the "Default deny rule" on the WAN interface.

After banging my head against the wall and cursing myself I realized the Host(s) Aliases caused the problem. After changing the use of an Alias to fixed IP-address in all NAT Port Forward rules everything worked again.

![alias](https://user-images.githubusercontent.com/2655169/34555816-137549e2-f134-11e7-8f97-1c5adbdaac55.jpg)

![synology_nas_dmz](https://user-images.githubusercontent.com/2655169/34555932-9e2782c6-f134-11e7-9964-46bfa981938c.jpg)

![nat_port_forward](https://user-images.githubusercontent.com/2655169/34555817-139275bc-f134-11e7-85ac-c3c12386126c.jpg)

When changing the IP-address 172.16.100.2 back to the Alias SYNOLOGY_NAS_DMZ,  NAT Port Forward no longer works and all traffic is blocked by the "Default deny rule".

Have I missed some fundamental changes regarding NAT Port Forward or is my setup broken after the update?

I can't believe this is a bug ...

**Comment by fichtner** [comment]
Since you are on -devel, did you reboot after the last update? The alias subsystem has been replaced. Maybe also this https://github.com/opnsense/core/issues/2042#issuecomment-354917351

**Comment by itheiss** [comment]
After switching on -devel I did a reboot for sure.

As pointed out by @brandonmreeves at #2042 I can confirm that the directory `/var/db/aliastables` doesn't exist after upgrade and reboot. So something definitely is wrong ...

**Comment by fichtner** [comment]
it's -devel after all :)

does it work if the directory is created manually as described? I'm assigning @AdSchellevis , he will know what to do.

**Comment by itheiss** [comment]
Creating and populating the directory manually brings Aliases back to life.

**Comment by fichtner** [comment]
One question, are you using the /var MFS option?

**Comment by itheiss** [comment]
Just realized that the directory is not cleaned up correctly when deleting an Alias. I have created an Alias TEST_LOCALHOST to check if everything is working after manually creating the directory. After deleting the Alias the files md5.txt and self.txt remain in the directory. 

Should I open a seperate issue for this?

![aliases_not_removed](https://user-images.githubusercontent.com/2655169/34557436-895c7076-f13a-11e7-8b58-ddfb02435c13.jpg)

**Comment by itheiss** [comment]
> One question, are you using the /var MFS option?

Never seen this. How can I check this?

**Comment by fichtner** [comment]
Easiest way is asking the system:

    # df -h

and posting the output.

**Comment by itheiss** [comment]
Ok. I could have thought it myself.

```
root@opnsense:~ # df -h
Filesystem         Size    Used   Avail Capacity  Mounted on
/dev/gpt/rootfs    215G    4.8G    193G     2%    /
devfs              1.0K    1.0K      0B   100%    /dev
devfs              1.0K    1.0K      0B   100%    /var/dhcpd/dev
```

**Comment by fichtner** [comment]
> Should I open a seperate issue for this?

Let's keep this in here, it's all related.

Ok, no /var MFS, thanks, that helps narrow it down :)

**Comment by AdSchellevis** [comment]
it's likely still resolving entries from test_localhost, the .self file should contain the lines that are processed already.

**Comment by itheiss** [comment]
@AdSchellevis Don't know what you mean with with `till resolving entries`. I have never used this Alias. Just created to check if everything is created inside `/var/db/aliastables` and then deleted the Alias.

Here is the output:
```
root@opnsense:/var/db/aliastables # cat TEST_LOCALHOST.self.txt
127.0.0.1
```

**Comment by AdSchellevis** [comment]
What was the content of the alias before you deleted it? maybe something else went wrong which crashed the script. 

Can you try to run the script manually and see if it fails somewhere?
`
/usr/local/opnsense/scripts/filter/update_tables.py
`

Thanks

**Comment by itheiss** [comment]
Ok here we go ...

**Created a new Alias named SIMPLE_TEST via GUI**
  ```
root@opnsense:/var/db/aliastables # ls -la SIMPLE_TEST.*
-rw-r-----  1 root  wheel  32 Jan  4 11:27 SIMPLE_TEST.md5.txt
-rw-r-----  1 root  wheel  10 Jan  4 11:27 SIMPLE_TEST.self.txt
-rw-r-----  1 root  wheel   9 Jan  4 11:27 SIMPLE_TEST.txt
  ```
_Content after creation:_
```
root@opnsense:/var/db/aliastables # cat SIMPLE_TEST.self.txt 
127.0.0.1
root@opnsense:/var/db/aliastables # cat SIMPLE_TEST.txt 
127.0.0.1
```

**Delete Alias named SIMPLE_TEST  via GUI**

_Before clicking "Apply changes" via GUI_
```
root@opnsense:/var/db/aliastables # ls -la SIMPLE_TEST.*
-rw-r-----  1 root  wheel  32 Jan  4 11:27 SIMPLE_TEST.md5.txt
-rw-r-----  1 root  wheel  10 Jan  4 11:27 SIMPLE_TEST.self.txt
```

The file SIMPLE_TEST.txt ist already removed!

Content before clicking "Apply changes" via GUI
```
root@opnsense:/var/db/aliastables # cat SIMPLE_TEST.self.txt 
127.0.0.1
```

_After clicking "Apply changes" via GUI_
```
root@opnsense:/var/db/aliastables # ls -la SIMPLE_TEST.*
-rw-r-----  1 root  wheel  32 Jan  4 11:27 SIMPLE_TEST.md5.txt
-rw-r-----  1 root  wheel  10 Jan  4 11:27 SIMPLE_TEST.self.txt
```
_Content after clicking "Apply changes" via GUI_
```
root@opnsense:/var/db/aliastables # cat SIMPLE_TEST.self.txt 
127.0.0.1
```

**Manually running the script `/usr/local/opnsense/scripts/filter/update_tables.py`**
```
root@opnsense:/var/db/aliastables # /usr/local/opnsense/scripts/filter/update_tables.py
root@opnsense:/var/db/aliastables # echo $?
0
```

_Content after manually running the script `/usr/local/opnsense/scripts/filter/update_tables.py`_
```
root@opnsense:/var/db/aliastables # cat SIMPLE_TEST.self.txt 
127.0.0.1
```

**Comment by AdSchellevis** [comment]
@itheiss I think I'm missing something here, the alias content is deleted before apply, which doesn't have to be a big issue as it's still loaded in the filter, but your issue started with the alias not being set for host type aliases. 

Is this issue reproducible? Normally using the new aliases it should load the tables, which are viewable  under Firewall->Diagnostics->pfTable in the new situation, then the nat rule should use the table.... question is, are we missing the nat rule or is the table empty. I'm not able to investigate this further today, but if you can check how I can reproduce this that would be very welcome.

**Comment by fichtner** [comment]
More patches went in.

**Comment by hirschferkel** [comment]
I still get this problem with 18.1 Production Series! Installed 18.1 nano-image and restored a backup from 17.7. But now, no defined Port range is working anymore, which belongs to an Alias. 
When I manually input all ports with IP-adresses it works. 

How can I get back my Aliases, now?

**Comment by AdSchellevis** [comment]
@hirschferkel there's not a lot of context in your question, but this might be related https://forum.opnsense.org/index.php?topic=7078.msg31436#new

**Comment by hirschferkel** [comment]
The context is the same as mentioned in the ticked in the beginning. We implemented Aliases to define port ranges and these are not working in the 18.1 stable version. That's all. If I ad every port manually not with an Alias but an IP it's working. I try to upload the patch, but it should ne fixed in a stable release, shouldn't it?

**Comment by AdSchellevis** [comment]
the fix will most likely be in the next release, but in the meantime you can use the patch

**Comment by hirschferkel** [comment]
The patch did not work.
> opened terminal in OSX
> logged in with ssh and root user
> entered opnsense-patch 57f51d2943
> reboot
 >> doesn't work with aliases.

**Comment by AdSchellevis** [comment]
Maybe another issue, we need more input to be able to help out. (preferably the exact definition of the rule and alias and a if possible the generated rules in /tmp/rules.debug for both versions to compare differences)

**Comment by hirschferkel** [comment]
Rules are just take all incoming connections and pass range of ports to one direction and corresponding ports.
Host is defined as Alias (but that's not the problem)
Port range is defined as Alias.

But what I found is, that old rules imported can not be edited!
I can edit a new rule but this one will not be selectable with NAT port forwarding!

Something has gone quite wrong here...
![bildschirmfoto 2018-01-31 um 13 42 38](https://user-images.githubusercontent.com/16016422/35623999-deb36b7a-068d-11e8-9b05-c622fad93a58.png)
![bildschirmfoto 2018-01-31 um 13 42 03](https://user-images.githubusercontent.com/16016422/35624000-ded496e2-068d-11e8-9519-2c9e15c31f75.png)

**Comment by hirschferkel** [comment]
It only works if I choose "pass" as an option, but i can't select new rules which are set to pass. I guess old rules loose their definition, as they can not be edited either. But i cannot set a portrange to be passed... that's wired.

**Comment by AdSchellevis** [comment]
Associated rules can't be changed from the firewall section, because the nat rule "owns" them. We have changed that a long time ago to prevent issues. 

Normally a save should automatically update the associated rule, but if there's something wrong with the import it might be better to copy the rule (then you can create a new filter rule) or manage the firewall rules manually (set to None). 

By my knowledge there haven't been changes in this area for 18.1.

**Comment by hirschferkel** [comment]
you're right, I missed that.

So if I autocreate a port forwarding, it will not work!
If I setup a rule one manually, it will not be available for a new port forwarding. So it won't work, anyway at the moment?

**Comment by AdSchellevis** [comment]
at my end the associated filter rules looks normal (the port alias is mentioned in both the rdr as the pass rule in /tmp/rules.debug).

**Comment by hirschferkel** [comment]
Great. At my end the filter rules look great, too. Actually they don't work. But I guess I wrote that already...

**Comment by hirschferkel** [comment]
After Updating to 18.1.1 it runs again. Obviously there were some more issues.
